File: api/csharp/ToolGood.TextFilter.Api/Impl/TextFilterProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ToolGood.TextFilter.Api.Datas.Texts;
using ToolGood.TextFilter.Api.Interfaces;
using Newtonsoft.Json;
using ToolGood.TextFilter.Api.Datas.Requests;

namespace ToolGood.TextFilter.Api.Impl
{
    public class TextFilterProvider : ProviderBase, ITextFilterProvider
    {
        public TextFilterProvider() : base(TextFilterConfig.Instance) { }
        public TextFilterProvider(TextFilterConfig textFilterConfig) : base(textFilterConfig) { }

        private const string TextFilterUrl = "/api/text-filter";
        private const string TextReplaceUrl = "/api/text-replace";
        private const string HtmlFilterUrl = "/api/html-filter";
        private const string HtmlReplaceUrl = "/api/html-replace";

        private const string JsonFilterUrl = "/api/json-filter";
        private const string JsonReplaceUrl = "/api/json-replace";
        private const string MarkdownFilterUrl = "/api/markdown-filter";
        private const string MarkdownReplaceUrl = "/api/markdown-replace";

        #region TextFilter TextReplace
        public TextFilterResult TextFilter(TextFilterRequest request)
        {
            return Filter(TextFilterUrl, request);
        }

        public Task<TextFilterResult> TextFilterAsync(TextFilterRequest request)
        {
            return FilterAsync(TextFilterUrl, request);
        }
        public TextReplaceResult TextReplace(TextReplaceRequest request)
        {
            return Replace(TextReplaceUrl, request);
        }

        public Task<TextReplaceResult> TextReplaceAsync(TextReplaceRequest request)
        {
            return ReplaceAsync(TextReplaceUrl, request);
        }

        #endregion

        #region HtmlFilter HtmlReplace


        public TextFilterResult HtmlFilter(TextFilterRequest request)
        {
            return Filter(HtmlFilterUrl, request);
        }

        public Task<TextFilterResult> HtmlFilterAsync(TextFilterRequest request)
        {
            return FilterAsync(HtmlFilterUrl, request);
        }

        public TextReplaceResult HtmlReplace(TextReplaceRequest request)
        {
            return Replace(HtmlReplaceUrl, request);
        }

        public Task<TextReplaceResult> HtmlReplaceAsync(TextReplaceRequest request)
        {
            return ReplaceAsync(HtmlReplaceUrl, request);
        }
        #endregion

        #region JsonFilter  JsonReplace
        public TextFilterResult JsonFilter(TextFilterRequest request)
        {
            return Filter(JsonFilterUrl, request);
        }

        public Task<TextFilterResult> JsonFilterAsync(TextFilterRequest request)
        {
            return FilterAsync(JsonFilterUrl, request);
        }

        public TextReplaceResult JsonReplace(TextReplaceRequest request)
        {
            return Replace(JsonReplaceUrl, request);
        }

        public Task<TextReplaceResult> JsonReplaceAsync(TextReplaceRequest request)
        {
            return ReplaceAsync(JsonReplaceUrl, request);
        }

        #endregion

        #region MarkdownFilter MarkdownReplace
        public TextFilterResult MarkdownFilter(TextFilterRequest request)
        {
            return Filter(MarkdownFilterUrl, request);
        }

        public Task<TextFilterResult> MarkdownFilterAsync(TextFilterRequest request)
        {
            return FilterAsync(MarkdownFilterUrl, request);
        }

        public TextReplaceResult MarkdownReplace(TextReplaceRequest request)
        {
            return Replace(MarkdownReplaceUrl, request);
        }

        public Task<TextReplaceResult> MarkdownReplaceAsync(TextReplaceRequest request)
        {
            return ReplaceAsync(MarkdownReplaceUrl, request);
        }


        #endregion

        private TextFilterResult Filter(string url, TextFilterRequest request)
        {
            if (string.IsNullOrEmpty(request.Txt)) { throw new ArgumentNullException(nameof(request.Txt)); }
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["txt"] = request.Txt;
            dict["skipBidi"] = request.SkipBidi;
            dict["onlyPosition"] = request.OnlyPosition;

            var json = JsonConvert.SerializeObject(dict);
            return PostContent<TextFilterResult>(url, json);
        }
        private Task<TextFilterResult> FilterAsync(string url, TextFilterRequest request)
        {
            if (string.IsNullOrEmpty(request.Txt)) { throw new ArgumentNullException(nameof(request.Txt)); }
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["txt"] = request.Txt;
            dict["skipBidi"] = request.SkipBidi;
            dict["onlyPosition"] = request.OnlyPosition;

            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<TextFilterResult>(url, json);
        }
        private TextReplaceResult Replace(string url, TextReplaceRequest request)
        {
            if (string.IsNullOrEmpty(request.Txt)) { throw new ArgumentNullException(nameof(request.Txt)); }
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["txt"] = request.Txt;
            dict["replaceChar"] = request.ReplaceChar;
            dict["reviewReplace"] = request.ReviewReplace;
            dict["skipBidi"] = request.SkipBidi;
            dict["contactReplace"] = request.ContactReplace;
            dict["onlyPosition"] = request.OnlyPosition;

            var json = JsonConvert.SerializeObject(dict);
            return PostContent<TextReplaceResult>(url, json);
        }
        private Task<TextReplaceResult> ReplaceAsync(string url, TextReplaceRequest request)
        {
            if (string.IsNullOrEmpty(request.Txt)) { throw new ArgumentNullException(nameof(request.Txt)); }
            Dictionary<string, object> dict = new Dictionary<string, object>();
            dict["txt"] = request.Txt;
            dict["replaceChar"] = request.ReplaceChar;
            dict["reviewReplace"] = request.ReviewReplace;
            dict["skipBidi"] = request.SkipBidi;
            dict["contactReplace"] = request.ContactReplace;
            dict["onlyPosition"] = request.OnlyPosition;

            var json = JsonConvert.SerializeObject(dict);
            return PostContentAsync<TextReplaceResult>(url, json);
        }







    }
}

File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Impl/TextFilterProvider.java
package toolgood.textfilter.api.Impl;

import java.util.Hashtable;
import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import toolgood.textfilter.api.TextFilterConfig;
import toolgood.textfilter.api.Datas.Requests.TextFilterRequest;
import toolgood.textfilter.api.Datas.Requests.TextReplaceRequest;
import toolgood.textfilter.api.Datas.Texts.TextFilterResult;
import toolgood.textfilter.api.Datas.Texts.TextReplaceResult;
import toolgood.textfilter.api.Interfaces.ITextFilterProvider;

public class TextFilterProvider extends ProviderBase implements ITextFilterProvider {

    public TextFilterProvider(TextFilterConfig textFilterConfig) {
        super(textFilterConfig);
    }

    private final String TextFilterUrl = "/api/text-filter";
    private final String TextReplaceUrl = "/api/text-replace";
    private final String HtmlFilterUrl = "/api/html-filter";
    private final String HtmlReplaceUrl = "/api/html-replace";

    private final String JsonFilterUrl = "/api/json-filter";
    private final String JsonReplaceUrl = "/api/json-replace";
    private final String MarkdownFilterUrl = "/api/markdown-filter";
    private final String MarkdownReplaceUrl = "/api/markdown-replace";

    @Override
    public TextFilterResult TextFilter(TextFilterRequest request) {
        String content = PostFilter(TextFilterUrl, request);

        Gson gson = new GsonBuilder().create();
        TextFilterResult result = gson.fromJson(content, TextFilterResult.class);
        return result;
    }

    @Override
    public TextReplaceResult TextReplace(TextReplaceRequest request) {
        String content = PostReplace(TextReplaceUrl, request);

        Gson gson = new GsonBuilder().create();
        TextReplaceResult result = gson.fromJson(content, TextReplaceResult.class);
        return result;
    }

    @Override
    public TextFilterResult HtmlFilter(TextFilterRequest request) {
        String content = PostFilter(HtmlFilterUrl, request);

        Gson gson = new GsonBuilder().create();
        TextFilterResult result = gson.fromJson(content, TextFilterResult.class);
        return result;
    }

    @Override
    public TextReplaceResult HtmlReplace(TextReplaceRequest request) {
        String content = PostReplace(HtmlReplaceUrl, request);

        Gson gson = new GsonBuilder().create();
        TextReplaceResult result = gson.fromJson(content, TextReplaceResult.class);
        return result;
    }

    @Override
    public TextFilterResult JsonFilter(TextFilterRequest request) {
        String content = PostFilter(JsonFilterUrl, request);

        Gson gson = new GsonBuilder().create();
        TextFilterResult result = gson.fromJson(content, TextFilterResult.class);
        return result;
    }

    @Override
    public TextReplaceResult JsonReplace(TextReplaceRequest request) {
        String content = PostReplace(JsonReplaceUrl, request);

        Gson gson = new GsonBuilder().create();
        TextReplaceResult result = gson.fromJson(content, TextReplaceResult.class);
        return result;
    }

    @Override
    public TextFilterResult MarkdownFilter(TextFilterRequest request) {
        String content = PostFilter(MarkdownFilterUrl, request);

        Gson gson = new GsonBuilder().create();
        TextFilterResult result = gson.fromJson(content, TextFilterResult.class);
        return result;
    }

    @Override
    public TextReplaceResult MarkdownReplace(TextReplaceRequest request) {
        String content = PostReplace(MarkdownReplaceUrl, request);

        Gson gson = new GsonBuilder().create();
        TextReplaceResult result = gson.fromJson(content, TextReplaceResult.class);
        return result;
    }

    private String PostFilter(String url, TextFilterRequest request) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("txt", request.Txt);
        dictionary.put("skipBidi", request.SkipBidi);
        dictionary.put("onlyPosition", request.OnlyPosition);
        return doPost(url, dictionary);
    }

    private String PostReplace(String url, TextReplaceRequest request) {
        Map<String, Object> dictionary = new Hashtable<String, Object>();
        dictionary.put("txt", request.Txt);
        dictionary.put("replaceChar", request.ReplaceChar);
        dictionary.put("reviewReplace", request.ReviewReplace);
        dictionary.put("skipBidi", request.SkipBidi);
        dictionary.put("contactReplace", request.ContactReplace);
        dictionary.put("onlyPosition", request.OnlyPosition);

        return doPost(url, dictionary);
    }

}
